### PR TITLE
[Merged by Bors] - feature(Analysis/Normed/Module/Dual): Lemma used in proof of Banach-Dieudonneé theorem

### PR DESCRIPTION
**Pull request by mans0954**
Adds a lemma implicit in the proof of Bourbaki, TVS IV.24, §5 Theorem 1 (Banach-Dieudonneé theorem)

(See also Lemma 12.2 from Conway, A Course in Functional Analysis for the Banach space version).

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/Normed/Module/Dual.lean [hunk 1]
```diff
@@ -234,6 +239,21 @@ theorem isBounded_polar_of_mem_nhds_zero {s : Set E} (s_nhd : s ∈ 𝓝 (0 : E)
     (((dualPairing 𝕜 E).flip.polar_antitone r_ball).trans <|
       polar_ball_subset_closedBall_div ha r_pos)
 
+theorem polar_singleton {a : E} : polar 𝕜 {a} = { x | ‖x a‖ ≤ 1 } := le_antisymm
```
Comment (j-loreaux) [inline Mathlib/Analysis/Normed/Module/Dual.lean:242]: Should this be `@[simp]`? I'm not sure, so feel free to argue if you disagree. If it were, then `mem_polar_singleton` is provable `by simp`.

File: Mathlib/Analysis/Normed/Module/Dual.lean [hunk 2]
```diff
@@ -234,6 +239,21 @@ theorem isBounded_polar_of_mem_nhds_zero {s : Set E} (s_nhd : s ∈ 𝓝 (0 : E)
...
+  (fun _ hx => hx _ rfl)
+  (fun x hx => (mem_polar_iff _ _).mpr (fun _ hb => by rw [mem_singleton_iff.mp hb]; exact hx))
+
+theorem mem_polar_singleton {a : E} (y : Dual 𝕜 E) : y ∈ polar 𝕜 {a} ↔ ‖y a‖ ≤ 1 := by
+  rw [polar_singleton]
+  exact mem_setOf
+
+theorem sInter_polar_finite_reciprocal_ball {𝕜 E : Type*} [RCLike 𝕜] [NormedAddCommGroup E]
```
Comment (j-loreaux) [inline Mathlib/Analysis/Normed/Module/Dual.lean:250]: or `sInter_polar_eq_closedBall`, if you want.
```suggestion
theorem sInter_polar_finite_reciprocal_closedBall {𝕜 E : Type*} [RCLike 𝕜] [NormedAddCommGroup E]
```

File: Mathlib/Analysis/LocallyConvex/Polar.lean [hunk 3]
```diff
@@ -111,6 +111,26 @@ theorem polar_weak_closed (s : Set E) : IsClosed[WeakBilin.instTopologicalSpace
...
+    ⋂₀ (B.polar '' { F | F.Finite ∧ F ⊆ U }) = B.polar U := by
+  apply le_antisymm _ _
+  · intro x hx
+    simp only [Set.sInter_image, Set.mem_setOf_eq, Set.mem_iInter, and_imp] at hx
+    intro a ha
+    rw [← mem_polar_singleton]
+    apply hx _ (Set.finite_singleton a) (Set.singleton_subset_iff.mpr ha)
+  · simp only [Set.sInter_image, Set.mem_setOf_eq, Set.le_eq_subset, Set.subset_iInter_iff,
+    and_imp]
+    exact fun F _ hF₂ => polar_antitone _ hF₂
```
Comment (j-loreaux) [inline Mathlib/Analysis/LocallyConvex/Polar.lean:132]: If you use `ext` to begin with, then you only need to do the `simp` once.
```suggestion
  ext x
  simp only [Set.sInter_image, Set.mem_setOf_eq, Set.mem_iInter, and_imp]
  refine ⟨fun hx a ha ↦ ?_, fun hx F _ hF₂ => polar_antitone _ hF₂ hx⟩
  simpa [mem_polar_singleton] using hx _ (Set.finite_singleton a) (Set.singleton_subset_iff.mpr ha)
```

File: Mathlib/Analysis/Normed/Module/Dual.lean (hunk 1 above)
Comment (mans0954) [reply Mathlib/Analysis/Normed/Module/Dual.lean]: On balance, I think so, but I may not be the best judge of what makes a good simp lemma.

**Comment by j-loreaux** [comment]
Thanks!

bors merge

**Pull request closed without merging**